src/com/seafile/seadroid2/account/Account.java

```diff
@@ -87,27 +76,27 @@ public boolean equals(Object obj) {
             return false;
 
         Account a = (Account)obj;
-        if (a.server == null || a.email == null)
+        if (a.server == null || a.email == null || a.token == null)
             return false;
 
         return a.server.equals(this.server) && a.email.equals(this.email);
     }
 
     public String getSignature() {
-        return email.substring(0, 4) + " " + hashCode();
+        return String.format("%s (%s)", getServerNoProtocol(), email);
     }
 
-    public String getFullSignature() {
-        return email + "@" + server;
+    public String getDisplayName() {
+        String server = Utils.stripSlashes(getServerHost());
+        return Utils.assembleUserName(email, server);
     }
 
-    public String getName() {
-        return email.substring(0, email.indexOf("@")) + "@" + getServerHost();
+    public android.accounts.Account getAndroidAccount() {
+        return new android.accounts.Account(getSignature(), ACCOUNT_TYPE);
     }
 
-    public String getDisplayName() {
-        String server = Utils.stripSlashes(getServerHost());
-        return Utils.assembleUserName(email, server);
+    public boolean hasValidToken() {
+        return token != null;
```

Comment:
why not use `public static boolean isEmpty (CharSequence str)` in TextUtils class